examples/gno.land/p/demo/gnolotto/gnolotto.gno

```diff
@@ -0,0 +1,100 @@
+package gnolotto
+
+import (
+	"std"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Ticket struct {
+	Numbers []int       // Holds the selected numbers for the lottery ticket
+	Owner   std.Address // Address of the ticket owner
+}
+
+type Lottery struct {
+	Tickets        []Ticket  // All tickets in the lottery
+	WinningNumbers []int     // Winning numbers after the draw
+	DrawTime       time.Time // Time of the draw
+	PrizePool      int64     // Total prize pool amount
+}
+
+// Intializes a new lottery instance with a specified draw time and prize pool
+func NewLottery(drawTime time.Time, prizePool int64) *Lottery {
+	return &Lottery{
+		DrawTime:  drawTime,
+		PrizePool: prizePool,
+		Tickets:   make([]Ticket, 0),
+	}
+}
+
+// Adds a new ticket to the lottery
+func (l *Lottery) AddTicket(numbers []int, owner std.Address) {
+	l.Tickets = append(l.Tickets, Ticket{Numbers: numbers, Owner: owner})
+}
+
+// Conducts the draw by generating 5 random numbers between 1 and 15
+func (l *Lottery) Draw() {
+	var blockHeight int64 = std.GetHeight()
+
+	l.WinningNumbers = nil
+	numbersMap := make(map[int]bool)
+
+	// Add variability to the pseudo-random number generation
+	var variabilityFactor int64 = 1
+
+	for len(l.WinningNumbers) < 5 {
+		simpleSeed := (blockHeight + variabilityFactor*251) % 233280
+		number := int(simpleSeed%15) + 1 // Ensure number is between 1 and 15
+
+		if !numbersMap[number] {
+			l.WinningNumbers = append(l.WinningNumbers, number)
+			numbersMap[number] = true
+		}
+
+		variabilityFactor += 13 // Adjusts for increased variability
+	}
+}
+
+// Itterate over all tickets to identify and return the addresses of the winners
+func (l *Lottery) CheckWinners() []std.Address {
+	var winningOwners []std.Address
+
+	for _, ticket := range l.Tickets {
+		matchCount := 0
+
+		for _, tNum := range ticket.Numbers {
+			for _, wNum := range l.WinningNumbers {
+				if tNum == wNum {
+					matchCount++
+					break
+				}
+			}
+		}
+
+		if matchCount == len(l.WinningNumbers) {
+			winningOwners = append(winningOwners, ticket.Owner)
+		}
+	}
+
```

Comment:
Please look into how this can be better optimized.
If you define a constant length for the amount of numbers in the tickets, you can have a single for loop iterating and comparing values, and breaking as soon as there is a mismatching pair.

Reply:
The problem with the solution you propose is that if the participant has all 5 numbers but they are not in the same order as the draw, he will be defined as a loser, which is not the expected result.

Reply:
Shouldn't a lottery ticket match the numbers _and_ the order of the winning set?

Reply:
You're right, the numbers must be in the same order. I'd forgotten this rule. here's the change in this [commit](https://github.com/gnolang/gno/pull/1850/commits/3c2409b0d52eacbfc627229e1468b691152e7e7d)